src/test/java/org/junitpioneer/jupiter/DefaultLocaleTests.java

```diff
@@ -0,0 +1,152 @@
+import org.junitpioneer.AbstractPioneerTestEngineTests;
+
+@DisplayName("DefaultLocale extension")
+class DefaultLocaleTests extends AbstractPioneerTestEngineTests {
+
+	private static Locale TEST_DEFAULT_LOCALE;
+	private static Locale DEFAULT_LOCALE_BEFORE_TEST;
+
+	@BeforeAll
+	static void globalSetUp() {
+		DEFAULT_LOCALE_BEFORE_TEST = Locale.getDefault();
+		TEST_DEFAULT_LOCALE = new Locale("custom");
+		Locale.setDefault(TEST_DEFAULT_LOCALE);
+	}
+
+	@AfterAll
+	static void globalTearDown() {
+		Locale.setDefault(DEFAULT_LOCALE_BEFORE_TEST);
+	}
+
+	@Nested
+	@DisplayName("applied on the method level")
+	class MethodLevelTests {
+
+		@Test
+		@DisplayName("does nothing when annotation is not present")
+		void testDefaultLocaleNoAnnotation() {
+			assertEquals(TEST_DEFAULT_LOCALE, Locale.getDefault());
+		}
+
+		@DefaultLocale(language = "en_EN")
+		@Test
+		@DisplayName("sets the default locale using a language")
+		void setsLanguage() {
+			assertEquals(new Locale("en_EN"), Locale.getDefault());
+		}
+
+		@DefaultLocale(language = "en", country = "EN")
+		@Test
+		@DisplayName("sets the default locale using a language and a country")
+		void setsLanguageAndCountry() {
+			assertEquals(new Locale("en", "EN"), Locale.getDefault());
+		}
+
+		@DefaultLocale(language = "en", country = "EN", variant = "gb")
+		@Test
+		@DisplayName("sets the default locale using a language, a country and a variant")
+		void setsLanguageAndCountryAndVariant() {
+			assertEquals(new Locale("en", "EN", "gb"), Locale.getDefault());
+		}
+	}
+
+	@Nested
+	@DisplayName("applied on the class level")
+	class ClassLevelTests {
+
+		@BeforeEach
+		void setUp() {
+			assertEquals(TEST_DEFAULT_LOCALE, Locale.getDefault());
+		}
+
+		@Test
+		@DisplayName("should execute tests with configured Locale")
+		void shouldExecuteTestsWithConfiguredLocale() {
+			ExecutionEventRecorder eventRecorder = executeTestsForClass(ClassLevelTestCase.class);
+
+			assertEquals(2, eventRecorder.getTestSuccessfulCount());
+		}
+
+		@AfterEach
+		void tearDown() {
+			assertEquals(TEST_DEFAULT_LOCALE, Locale.getDefault());
+		}
+	}
+
+	@DefaultLocale(language = "fr", country = "FR")
+	static class ClassLevelTestCase {
+
+		@Test
+		void shouldExecuteWithClassLevelLocale() {
+			assertEquals(new Locale("fr", "FR"), Locale.getDefault());
+		}
+
+		@Test
+		@DefaultLocale(language = "de", country = "DE")
+		void shouldBeOverriddenWithMethodLevelLocale() {
+			assertEquals(new Locale("de", "DE"), Locale.getDefault());
+		}
+	}
+
+	@Nested
+	@DisplayName("when configured incorrect")
+	class ConfigurationFailureTests {
+
+		@Test
+		@DisplayName("should fail when variant is set but country is not on method level")
+		void shouldFailWhenVariantIsSetButCountryIsNotOnMethodLevel() {
+			ExecutionEventRecorder eventRecorder = executeTestsForClass(MethodLevelInitializationFailureTestCase.class);
+
+			assertEquals(1, eventRecorder.getTestFailedCount());
```

Comment:
I wonder whether there would be much value in asserting the concrete error message or at least the exception type. What do you think?

Reply:
If think assertion the exception type is a good idea. I've added that. Checking exception messages is not a good idea IMHO, since that can change over time to provide more context and there is really no value in checking that text.

OTOH this is now a catch all for `ExtensionConfigurationExceptions`... so maybe it would be a good idea to check the test. I'm indifferent here. I think I'm fine with checking the type for now.